### PR TITLE
[refactor] remove PredicateColumn

### PR DESCRIPTION
**Pull request by wangbo**
# Proposed changes

remove PredicateColumn

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/bitshuffle_page.h [hunk 1]
```diff
@@ -392,7 +392,14 @@ class BitShufflePageDecoder : public PageDecoder {
 
         size_t max_fetch = std::min(*n, static_cast<size_t>(_num_elements - _cur_index));
 
-        dst->insert_many_fix_len_data(get_data(_cur_index), max_fetch);
+        // todo(wb) remove this branch after the data format is completely unified
+        if constexpr (Type == OLAP_FIELD_TYPE_DATE) {
+            dst->insert_many_date(get_data(_cur_index), max_fetch);
+        } else if constexpr (Type == OLAP_FIELD_TYPE_DATETIME) {
```
Comment (yiguolei) [inline be/src/olap/rowset/segment_v2/bitshuffle_page.h:398]: 我倾向在 column 中做这种判断，比如我们可以调用insert fixed length data， 但是在这个函数里我们判断一下column的类型，做if else 判断

File: be/src/olap/schema.cpp [hunk 2]
```diff
@@ -127,48 +126,45 @@ vectorized::IColumn::MutablePtr Schema::get_predicate_column_nullable_ptr(FieldT
 vectorized::IColumn::MutablePtr Schema::get_predicate_column_ptr(FieldType type) {
     switch (type) {
     case OLAP_FIELD_TYPE_BOOL:
-        return doris::vectorized::PredicateColumnType<bool>::create();
+        return doris::vectorized::ColumnVector<doris::vectorized::UInt8>::create();
```
Comment (yiguolei) [inline be/src/olap/schema.cpp:129]: move all these method to TabletColumn

File: be/src/olap/bloom_filter_predicate.h [hunk 3]
```diff
@@ -81,18 +80,42 @@ class BloomFilterColumnPredicate : public ColumnPredicate {
                     new_size += _specific_filter->find_uint32_t(dict_col->get_hash_value(idx));
                 }
             }
+        } else if (column.is_column_string()) {
+            if constexpr (std::is_same_v<file_type, StringValue>) {
```
Comment (yiguolei) [inline be/src/olap/bloom_filter_predicate.h:84]: file_type  or fieldType?

File: be/src/olap/rowset/segment_v2/bitshuffle_page.h (hunk 1 above)
Comment (wangbo) [reply be/src/olap/rowset/segment_v2/bitshuffle_page.h:398]: 1 分支放这的好处是可以用constexpr
2 如果要放column里，如果不加模板，单纯用if判断的话，就没法用constexpr
如果要用模板的话，那只能给column加个模板，这个改动太大了，所有用到column的地方都得加一个模板标识<>
另外模板应该没法给insert方法加，因为那个是虚函数
3 这个问题我觉得从最终方案考虑，如果最终我们决定对于date类型还是用columnvector保存，那么一定会出现一个columnvector保存多种类型的情况，那么就需要columnvector能够保存一个类型的模板，这个问题就解决了。
如果我们计划加一个新的类型比如dateColumn用于保存date类型，那么我建议保持现状，暂时先用这个分支

**Pull request closed without merging**